src/main/java/bio/terra/service/dataset/flight/delete/DeleteDatasetPrimaryDataStep.java

```diff
@@ -49,6 +56,21 @@ public StepResult doStep(FlightContext context) {
             fileDao.deleteFilesFromDataset(dataset, fireStoreFile -> gcsPdao.deleteFile(fireStoreFile));
         }
 
+        // this fault is used by the DatasetConnectedTest > testOverlappingDeletes
+        if (configService.testInsertFault(ConfigEnum.DATASET_DELETE_LOCK_CONFLICT_STOP_FAULT)) {
+            try {
+                logger.info("DATASET_DELETE_LOCK_CONFLICT_STOP_FAULT");
+                while (!configService.testInsertFault(ConfigEnum.DATASET_DELETE_LOCK_CONFLICT_CONTINUE_FAULT)) {
+                    logger.info("Sleeping for CONTINUE FAULT");
+                    TimeUnit.SECONDS.sleep(5);
+                }
+                logger.info("DATASET_DELETE_LOCK_CONFLICT_CONTINUE_FAULT");
+            } catch (InterruptedException intEx) {
```

Comment:
We should stop catching InterruptedException for anything used inside steps. They should just let those fly so the flight can be shutdown when the pod is getting shut down.

Reply:
The reason I catch InterruptedException here is that sleep() can throw it. Do you mean that I should let the doStep() method throw an InterruptedException? That would mean adding it to the method signature, which I think first requires adding it to the method signature in the base class bio.terra.stairway.Step.

Reply:
One way we gracefully shutdown Stairway before Pod termination is to terminate the thread pool. That delivers InterruptedException to all of the threads running Steps. Stairway catches that and marks the flights as READY so another Stairway instance will pick them up.

So we need to not mask the InterruptedException processing anymore. I know I have written exactly that code fragment many times. There is cleanup to do!

Reply:
Understood. Since this pattern is used in other places, I propose we handle this cleanup in a separate ticket. I will make one and would be happy to work on it!

Reply:
Added https://broadworkbench.atlassian.net/browse/DR-914